src/models/transactions/common.ts

```diff
@@ -0,0 +1,136 @@
+import { ValidationError } from "../../common/errors"
+import { Memo, Signer } from "../common"
+
+const transactionTypes = [
+    "AccountSet",
+    "AccountDelete",
+    "CheckCancel",
+    "CheckCash",
+    "CheckCreate",
+    "DepositPreauth",
+    "EscrowCancel",
+    "EscrowCreate",
+    "EscrowFinish",
+    "OfferCancel",
+    "OfferCreate",
+    "Payment",
+    "PaymentChannelClaim",
+    "PaymentChannelCreate",
+    "PaymentChannelFund",
+    "SetRegularKey",
+    "SignerListSet",
+    "TicketCreate",
+    "TrustSet",
+]
+
+const isMemo = (obj: {Memo: Memo}): boolean => {
+    const memo = obj.Memo
+    const size = Object.keys(memo).length
+    const validData = memo.MemoData === undefined 
+        || typeof memo.MemoData === 'string'
+    const validFormat = memo.MemoFormat === undefined 
+        || typeof memo.MemoData === 'string'
+    const validType = memo.MemoType === undefined
+        || typeof memo.MemoType === 'string'
+
+    const otherKeys = Object.keys(memo).every((key: string) => {
+        return key === "MemoFormat" 
+            || key === "MemoData"
+            || key === "MemoType"
+    })
```

Comment:
Is there a better way to do this? Maybe a helper function that makes sure there aren't extra keys in an object would be useful (idk if that's possible) - could be used in `validate` methods.

Reply:
I don't think this is a bad use of `every`. I don't love it, and am open to other ideas.

Maybe a helper like 

```ts
onlyHasFields(obj: Record<string, unkown>, fields: Array<string>) bool {
    return Object.key(obj).every((key:string) => fields.includes(key)
}
```

thoughts?